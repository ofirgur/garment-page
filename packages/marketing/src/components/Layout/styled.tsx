import styled from 'styled-components';

export const LayoutStyled = styled.div`
  color: ${props => props.theme.colors.persianGreen};
`;
