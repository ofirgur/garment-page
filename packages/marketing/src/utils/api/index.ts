import axios, { AxiosRequestConfig } from 'axios';
import * as MockAdapter from 'axios-mock-adapter';
import { MockConfig } from './type';

/* 
There is also a built-in environment variable called NODE_ENV. 
You can read it from process.env.NODE_ENV. When you run npm start, 
it is always equal to 'development', when you run npm test it is always equal to 'test', 
and when you run npm run build to make a production bundle, it is always equal to 'production'. 
You cannot override NODE_ENV manually. 
This prevents developers from accidentally deploying a slow development build to production.
*/

export const USE_MOCK = process.env.NODE_ENV !== 'production';
export const DEFAULT_BASE_URL = 'https://jsonplaceholder.typicode.com';
export const DEFAULT_MOCK_CONFIG = {
  mockDelay: 2000,
  mockResponse: {
    status: 200,
    data: 'createAPIAction function: default mockConfig',
  },
};

axios.defaults.baseURL = DEFAULT_BASE_URL;

export const createAPIAction =
  (
    requestConfig: AxiosRequestConfig,
    getMockConfig?: () => Promise<MockConfig>,
  ) =>
  async () => {
    if (USE_MOCK) {
      const { mockDelay, mockResponse } =
        (await getMockConfig()) || DEFAULT_MOCK_CONFIG;
      // This sets the mock adapter on the default instance
      const mock = new MockAdapter(axios, { delayResponse: mockDelay });
      // Match ALL requests
      mock.onAny().reply(mockResponse.status, mockResponse.data);
    }

    return axios(requestConfig).then(res => res.data);
  };

export const lazyMock =
  (importPromise: () => Promise<{ default: MockConfig }>) => async () => {
    const module = await importPromise();

    return module.default;
  };
