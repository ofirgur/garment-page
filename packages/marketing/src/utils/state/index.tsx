import * as React from 'react';
import { PageStateContextInterface, PageState } from './type';

// state management context
const PageStateContext = React.createContext<PageStateContextInterface | null>(
  null,
);

// custome hook to get the state of the app, getter & setter,
// return { pageState, setPageState }, see Ln 32, Col 38
export const usePageState = () => {
  const context = React.useContext(PageStateContext);
  if (!context) {
    throw new Error(
      'usePageState must be used within a PageStateContextProvider',
    );
  }

  return context;
};

// context frovider for app state, should wrap the Root of the app
// it's a component with local state,
// it shares it's state (getter & setter) with the returned PageStateContext.Provider (context provider)
type PageStateContextProviderProps = {
  initialState: PageState;
  children: React.ReactElement;
};

export const PageStateContextProvider = (
  props: PageStateContextProviderProps,
) => {
  const { initialState } = props;
  const [pageState, regularSetPageState] = React.useState(initialState);
  const setPageState = (newState: Partial<PageState>) => {
    regularSetPageState((prevState: PageState) => ({
      ...prevState,
      ...newState,
    }));
  };

  const value = React.useMemo(() => ({ pageState, setPageState }), [pageState]);
  return <PageStateContext.Provider value={value} {...props} />;
};
