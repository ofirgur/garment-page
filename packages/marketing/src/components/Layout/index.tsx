import * as React from 'react';

import { LayoutStyled } from './styled';
import { LayoutProps } from './type';

const Layout = ({ children }: LayoutProps) => {
  return <LayoutStyled>{children}</LayoutStyled>;
};

export default Layout;
