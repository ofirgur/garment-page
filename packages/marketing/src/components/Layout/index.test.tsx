import * as React from 'react';
import { render } from '../../utils/test';

import Layout from './index';

describe('Layout', () => {
  it('Should render correctly', () => {
    const { container } = render(<Layout />);
    expect(container).toMatchSnapshot();
  });
});
