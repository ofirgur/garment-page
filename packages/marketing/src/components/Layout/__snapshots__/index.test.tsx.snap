// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB cIOqyG"
  />
</div>
`;
